base/src/commonMain/kotlin/flow/BaseFlowable.kt

```diff
@@ -22,21 +22,33 @@ import kotlinx.coroutines.channels.BroadcastChannel
 import kotlinx.coroutines.channels.ConflatedBroadcastChannel
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onStart
 
-open class BaseFlowable<T>(private val channel: BroadcastChannel<T> = ConflatedBroadcastChannel()) : Flowable<T> {
+open class BaseFlowable<T>(private val channelFactory: () -> BroadcastChannel<T> = {ConflatedBroadcastChannel()}) : Flowable<T> {
+
+    private var channel = lazy {  channelFactory() }
+    private var flowing: Boolean = false
```

Comment:
Why do we need this flag?

Reply:
Shouldnt be able to set until flowing to prevent invalid State transitions

Reply:
E.g. make sure this is a cold stream, which Broadcast channel is not by default.
Nevertheless I made a fix related to this